apps/sms/views/shared/js/utils.js

```diff
@@ -473,6 +473,25 @@
       });
       return parsed;
     },
+    url(base, params) {
+      if (base.indexOf('?') === -1) {
+        base += '?';
+      } else {
+        base += '&';
+      }
+
+      for (var key in params) {
+        if (params[key] == null) { // null or undefined
+          continue;
+        }
+
+        base +=
+          encodeURIComponent(key) + '=' +
+          encodeURIComponent(params[key]) + '&';
```

Comment:
No I don't think it's over-engineered, sounds good enough for me. I was thinking about something like this as well